src/ast/expression.rs

```diff
@@ -466,31 +466,39 @@ impl<'a> Comparable<'a> for Expression<'a> {
     where
         T: Into<Expression<'a>>,
     {
-        Compare::JsonCompare(JsonCompare::ArrayBeginsWith(Box::new(self), Box::new(item.into())))
+        let array_starts_with: Expression = json_extract_first_array_elem(self).into();
+
+        Compare::Equals(Box::new(array_starts_with), Box::new(item.into()))
     }
 
     #[cfg(all(feature = "json", any(feature = "postgresql", feature = "mysql")))]
     fn json_array_not_begins_with<T>(self, item: T) -> Compare<'a>
     where
         T: Into<Expression<'a>>,
     {
-        Compare::JsonCompare(JsonCompare::ArrayNotBeginsWith(Box::new(self), Box::new(item.into())))
+        let array_starts_with: Expression = json_extract_first_array_elem(self).into();
+
+        Compare::NotEquals(Box::new(array_starts_with), Box::new(item.into()))
     }
 
     #[cfg(all(feature = "json", any(feature = "postgresql", feature = "mysql")))]
     fn json_array_ends_into<T>(self, item: T) -> Compare<'a>
     where
         T: Into<Expression<'a>>,
     {
-        Compare::JsonCompare(JsonCompare::ArrayEndsInto(Box::new(self), Box::new(item.into())))
+        let array_ends_into: Expression = json_extract_last_array_elem(self).into();
+
+        Compare::Equals(Box::new(array_ends_into), Box::new(item.into()))
     }
 
     #[cfg(all(feature = "json", any(feature = "postgresql", feature = "mysql")))]
     fn json_array_not_ends_into<T>(self, item: T) -> Compare<'a>
     where
         T: Into<Expression<'a>>,
     {
-        Compare::JsonCompare(JsonCompare::ArrayNotEndsInto(Box::new(self), Box::new(item.into())))
+        let array_ends_into: Expression = json_extract_last_array_elem(self).into();
+
+        Compare::NotEquals(Box::new(array_ends_into), Box::new(item.into()))
```

Comment:
The private functions are used here 👆.

Instead of having special `JsonCompare`s, we rely on the existing `Compare::Equals` & `Compare::NotEquals`.

The private functions help us _wrap_ the left side of the equal into whatever's needed to get the first/last item of an array.